rotkehlchen/chain/polygon_pos/modules/giveth/decoder.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.decoding.giveth.constants import CPT_DETAILS_GIVETH, CPT_GIVETH
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.utils.misc import bytes_to_address
+
+from .constants import DONATION_MADE_TOPIC, GIVETH_DONATION_CONTRACT_ADDRESS
+
+if TYPE_CHECKING:
+    from rotkehlchen.types import ChecksumEvmAddress
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class GivethDecoder(DecoderInterface):
+
+    def _decode_donation_events(self, context: DecoderContext) -> DecodingOutput:
+        if (
+                context.tx_log.topics[0] != DONATION_MADE_TOPIC or
+                not self.base.is_tracked(bytes_to_address(context.tx_log.topics[1]))
+        ):
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_received = asset_normalized_value(
+            amount=int.from_bytes(context.tx_log.data[:32]),
+            asset=(token_received := self.base.get_token_or_native(bytes_to_address(context.tx_log.topics[2]))),  # noqa: E501
+        )
+        for event in context.decoded_events:
```

Comment:
Just like with gitcoin you should do both sides. PErhaps you are the donator. In which case you should be donating and not receiving a donation.